core/src/main/java/org/apache/calcite/plan/RelOptUtil.java

```diff
@@ -829,6 +829,9 @@ public static RelNode createNullFilter(
    * instead, create a projection with the input of {@code rel} and the new
    * cast expressions.
    *
+   * <p>The field count of the desired row type and the row type to be converted
+   * should be equal. Otherwise, it would throw an exception.
+   *
```

Comment:
This can be placed in an "@throws" comment?

Reply:
+1 for `@throws`

Reply:
-0 for `throws`. A clear description is sufficient. Redundancy is worse.